### ISSUE TITLE
include_url not working - error in enc2utf8(x)

### ISSUE THREAD

**Issue by shharrison** [yihui/knitr#1820]
Hello, I'm having a similar issue to this post from [last year](https://github.com/yihui/knitr/issues/1691) . I am trying to include an iframe in an RMarkdown document.

````{r}
knitr::include_url("https://shharrison.carto.com/builder/f70b9be5-c863-4787-98ac-ba359cae9133/embed")
````
returns this: 

```{r}
Error in enc2utf8(x) : argument is not a character vector
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('knitr')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/knitr')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
I can reproduce but only when running chunk from RStudio IDE. 
When knitting `knitr::knit` or `rmarkdown::render` the document in `html_document` format, it results ok. Can you confirm this behavior too ?

**Comment by cderv** [comment]
After looking a bit, I think this is an issue with RStudio IDE.
The traceback I got is this one
![image](https://user-images.githubusercontent.com/6791940/76711122-8281c580-670d-11ea-987a-4b77c9cd0558.png)

It seems that for `include_url` and `include_app`, `print.knit_asis` is used by RStudio, and the error is thrown in [here](https://github.com/rstudio/rstudio/blob/28a5e3ff51462e836e4cd002507853ec92de87ee/src/cpp/session/modules/NotebookHtmlWidgets.R#L76) from RStudio IDE code.  

 `include_url` and `include_app` create special objects of class `knit_embed_url` and I think there is something missing to print the result of those call by the run current chunk button. 

### Reprex 

Paste this in a new Rmd document and try to run chunk using green arrow from the IDE. 
````r
---
title: "test"
output: html_document
---

```{r}
knitr::include_url("https://shharrison.carto.com/builder/f70b9be5-c863-4787-98ac-ba359cae9133/embed")
```
```{r}
knitr::include_app("https://gabrielvazdemelo.shinyapps.io/shiny_barragens_site/", height = "700px")
```
```

Tested on last RStudio Release and Preview. I think it worth opening an issue in RStudio IDE repo 🤔

**Comment by yihui** [comment]
Since this is an RStudio IDE issue, I'm going to close this one in **knitr**. @shharrison Please keep an eye on https://github.com/rstudio/rstudio/issues/6476. Thanks!

**Comment by cderv** [comment]
Duplicated by #2012 where we decided to fix it in #2015 so this should no more happen
